pkg/skaffold/schema/v1alpha2/config.go

```diff
@@ -238,17 +241,31 @@ func (c *SkaffoldConfig) setDefaultWorkspaces() {
 	}
 }
 
-func (c *SkaffoldConfig) expandKanikoSecretPath() error {
-	if c.Build.KanikoBuild == nil || c.Build.KanikoBuild.PullSecret == "" {
-		return nil
+func (c *SkaffoldConfig) setDefaultNamespace() {
+	if c.Build.KanikoBuild == nil {
+		return
+	}
+	if c.Build.KanikoBuild.Namespace == "" {
+		c.Build.KanikoBuild.Namespace = "default"
 	}
+}
 
-	absPath, err := homedir.Expand(c.Build.KanikoBuild.PullSecret)
-	if err != nil {
-		return fmt.Errorf("unable to expand pullSecret %s", c.Build.KanikoBuild.PullSecret)
+func (c *SkaffoldConfig) setDefaultSecret() error {
+	if c.Build.KanikoBuild == nil {
+		return nil
 	}
+	if c.Build.KanikoBuild.PullSecret != "" {
+		absPath, err := homedir.Expand(c.Build.KanikoBuild.PullSecret)
+		if err != nil {
+			return fmt.Errorf("unable to expand pullSecret %s", c.Build.KanikoBuild.PullSecret)
+		}
 
-	c.Build.KanikoBuild.PullSecret = absPath
+		c.Build.KanikoBuild.PullSecret = absPath
+		return nil
+	}
+	if c.Build.KanikoBuild.PullSecretName == "" {
+		c.Build.KanikoBuild.PullSecret = "kaniko-secret"
```

Comment:
maybe add this as a constant somewhere, since its references a few times in different files

Reply:
Done

Reply:
Doesn't look like you used the constant here